### PR TITLE
network: make the pubsub topic whitelist slot-aware (#2968 item 3)

### PR DESCRIPTION
**Pull request by momosh-ssv**
Item 3 of #2968.

`commons.Topics` gated the Alan topic set on the coarse `BooleFork()` epoch snapshot while subscriptions derive per-slot from the transition-window predicate — so a node **started inside the ~34-slot subsequent window** whitelisted Boole-only yet legitimately subscribes to Alan topics for late pre-fork committee traffic, working only incidentally because `topicsCtrl.Subscribe` self-registers.

`Topics` now takes the evaluation slot explicitly and includes the Alan set whenever Alan traffic can still be valid at that slot (`!BooleForkAtSlot(slot) || InBooleTransitionWindow(slot)`), always including Boole; the sole caller (`NewPubSub`) passes `EstimatedCurrentSlot()`. The startup-snapshot design is unchanged — the snapshot just now matches the subscription policy at the same instant.

Deep review notes (passed): the include-Alan condition is logically identical to all three per-slot subscription-derivation sites in `network/p2p`; the whitelist gates subscriptions/peer-announcements (mesh membership), not message validation, and the widening is confined to the subsequent window where the fork design requires Alan traffic to flow; a node started past the window gets Boole-only with no Alan leak (`Boole==0` guard verified). Five-phase table test covers pre-fork steady state, prior window, subsequent window (the bug), the exact first slot past the window (the boundary that closes Alan), and well post-fork.

### CODE REVIEW

File: network/commons/subnets_test.go [hunk 1]
```diff
@@ -210,6 +214,113 @@ func TestDiffSubnets(t *testing.T) {
 	require.Equal(t, 128-62, added.ActiveCount()+removed.ActiveCount())
 }
 
+// TestTopics covers the four Boole-transition phases the pubsub startup whitelist must track in
```
Comment (iurii-ssv) [inline network/commons/subnets_test.go]: Nit: the comment says *four* Boole-transition phases, but `TestTopics` has five subtests — the `first slot past the subsequent window is Boole-only` boundary case is a fifth. Suggest updating to *five* to match the PR description. (Minor: these are sequential `t.Run` subtests rather than a table-driven test, if you also want to reconcile the description's "table test" wording.)

File: network/commons/subnets_test.go [hunk 2]
```diff
@@ -210,6 +214,113 @@ func TestDiffSubnets(t *testing.T) {
...
+		for topic := range booleTopicsSet(cfg.SSV.Name) {
+			want[topic] = struct{}{}
+		}
+		require.Equal(t, want, got)
+	})
+
+	t.Run("first slot past the subsequent window is Boole-only", func(t *testing.T) {
+		cfg := booleCfg(cur + 1)
+		// One slot past the window end [F, F+SlotsPerEpoch+2) — the boundary that closes Alan.
+		slot := cfg.FirstSlotAtEpoch(cur+1) + phase0.Slot(cfg.SlotsPerEpoch) + 2
```
Comment (iurii-ssv) [inline network/commons/subnets_test.go:305]: Nit: this `+ 2` — and the `[F, F+SlotsPerEpoch+2)` in the comment just above — hard-codes the `booleSubsequentWindowLateSlots` margin, which is unexported in `networkconfig` and so can't be referenced from here. It's self-guarding: if that margin ever changed, the `require.False(cfg.InBooleTransitionWindow(slot), ...)` on the next lines would fail loudly rather than pass silently. Still, a one-line comment tying this literal back to `booleSubsequentWindowLateSlots` would save a future reader the cross-package hunt.

**Review by iurii-ssv — DISMISSED** [review]

File: network/commons/subnets_test.go (hunk 1 above)
Comment (momosh-ssv) [reply network/commons/subnets_test.go]: Fixed in 86489a4ee, the comment now says five and names the past-the-window boundary case explicitly.

File: network/commons/subnets_test.go (hunk 2 above)
Comment (momosh-ssv) [reply network/commons/subnets_test.go:305]: Fixed in 86489a4ee.
Added a comment tying the +2 to networkconfig's unexported booleSubsequentWindowLateSlots and noting the InBooleTransitionWindow assertion is the loud tripwire if the margin changes.

 Kept the literal rather than exporting the constant, seems not worth widening the API for a test.

**Review by y0sher — APPROVED** [review]

**Review by ljuba-ssv — APPROVED** [review]
